lib/job-iteration/integrations.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module JobIteration
+  module Integrations # @private
+    LoadError = Class.new(StandardError)
+
+    autoload :Sidekiq, "job-iteration/integrations/sidekiq"
+    autoload :Resque, "job-iteration/integrations/resque"
+
+    extend self
+
+    def load(job)
+      camelized_name = job.class.queue_adapter_name.camelize
+      object = const_get(camelized_name)
```

Comment:
> Interesting. Recommending that users monkey patch your library feels off to me, there must be better ways to provide that configuration. How do you feel about a lookup table from `queue_adapter_name` to interruption adapter that users can add to?

Maybe I was pattern matching too much on Active Job/GoodJob, your suggestion makes sense to me.

> That way, queue adapters that trigger the Rails bug can just register for `"class"` and we don't have to work around it here. (I [fixed the bug](https://github.com/rails/rails/pull/47995), btw 🙂, although that only helps for Rails 7.1 and on)

Awesome 🙌 it would be useful to backport to 7.0 IMO, for those of us not living on Rails main 😛 I don't think it returning `"module"` is useful/expected.